src/Microsoft.Graph.Core/Requests/BatchRequestBuilder.cs

```diff
@@ -4,42 +4,67 @@
 
 namespace Microsoft.Graph.Core.Requests
 {
-    using System.Collections.Generic;
+    using Microsoft.Kiota.Abstractions;
+    using System;
+    using System.Net.Http;
+    using System.Threading.Tasks;
 
     /// <summary>
     /// The type BatchRequestBuilder
     /// </summary>
-    public class BatchRequestBuilder: BaseRequestBuilder, IBatchRequestBuilder
+    public class BatchRequestBuilder
     {
         /// <summary>
         /// Constructs a new BatchRequestBuilder.
         /// </summary>
-        /// <param name="requestUrl">The URL for the built request.</param>
-        /// <param name="client">The <see cref="IBaseClient"/> for handling requests.</param>
-        public BatchRequestBuilder(
-            string requestUrl, 
-            IBaseClient client) 
-            : base(requestUrl, client)
+        /// <param name="requestAdapter">The request adapter to use to execute the requests.</param>
+        public BatchRequestBuilder(IRequestAdapter requestAdapter)
         {
+            _ = requestAdapter ?? throw new ArgumentNullException(nameof(requestAdapter));
+            UrlTemplate = "{+baseurl}/$batch";
+            RequestAdapter = requestAdapter;
         }
 
         /// <summary>
-        /// Builds the request.
+        /// Url template to use to build the URL for the current request builder
         /// </summary>
-        /// <returns>The built request.</returns>
-        public IBatchRequest Request()
+        internal string UrlTemplate { get; set; }
+
+        /// <summary>
+        /// The request adapter to use to execute the requests.
+        /// </summary>
+        internal IRequestAdapter RequestAdapter { get; set; }
+
+        /// <summary>
+        /// Sends out the <see cref="BatchRequestContent"/> using the POST method
+        /// </summary>
+        /// <param name="batchRequestContent">The <see cref="BatchRequestContent"/> for the request</param>
+        /// <returns></returns>
+        public async Task<BatchResponseContent> PostAsync(BatchRequestContent batchRequestContent)
         {
-            return this.Request(null);
+            _ = batchRequestContent ?? throw new ArgumentNullException(nameof(batchRequestContent));
+            var requestInfo = await CreatePostRequestInformationAsync(batchRequestContent);
+            var responseHandler = new NativeResponseHandler();
+            await this.RequestAdapter.SendNoContentAsync(requestInfo, responseHandler);
```

Comment:
out of curiosity, is there any specific reason why you're using the response handler instead of the provided deserialziation?

Reply:
For the batch scenario, we typically return the native response object i.e. `HttpReponseMessage` and wrap it around the `BatchResponseContent` which would now do the job unwrapping the collection of responses received in the batch. Since in this scenario we need an instance of `HttpReponseMessage`, using the response handler is the way to go.

Reply:
thanks for clarifying.
QQ: how do you find the handler pattern from a usability perspective?

Reply:
I think there's some great potential for it. 
Taking the example from the `NativeResponseHandler`, I'm realizing we could simply write a response handler that gives one back a `GraphResponse` object rather than the current method of having to generate parallel methods as we do [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet/blob/6f133fc66b313b5355913e568a426bce25bfa8ad/src/Microsoft.Graph/Generated/requests/AadUserConversationMemberRequest.cs#L59) saving us a bit of work and then the user just passes in a parameter based on what/how the response is done. 

I am unsure how I feel though about having to do this with the `NativeResponseHandler` 
```cs
responseHandler.Value as HttpResponseMessage;
```
and maybe we could refactor the `NativeResponseHandler` to be `NativeResponseHandler<T>` so that the Value  property is already is strongly typed from the get go.